src/test/java/hudson/scm/JEP200Test.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.scm;
+
+import com.google.common.base.Predicate;
+import hudson.remoting.ClassFilter;
+import hudson.util.VersionNumber;
+import jenkins.model.Jenkins;
+import jenkins.security.ClassFilterImpl;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.reflections.ReflectionUtils;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Set;
+import static java.lang.reflect.Modifier.*;
+
+/**
+ * Tests compatibility of the plugin with JEP-200 in Jenkins 2.102+
+ * @author Oleg Nenashev
+ */
+public class JEP200Test {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    @Issue("JENKINS-49574") // Fails on Jenkins 2.102+
+    public void checkClassesForDefaultRemotingBlacklist() {
+        ClassFilter cf;
+        try {
+            Class<?> cfclazz = Class.forName("jenkins.security.ClassFilterImpl");
+            Constructor c = cfclazz.getDeclaredConstructor();
+            c.setAccessible(true);
+            cf = (ClassFilter) c.newInstance();
+        } catch (ClassNotFoundException | IllegalAccessException | NoSuchMethodException | InvocationTargetException | InstantiationException ex) {
+            if (new VersionNumber(Jenkins.VERSION).isNewerThan(new VersionNumber("2.101"))) {
+                throw new AssertionError("Jenkins version is newer than 2.101, jenkins.security.ClassFilterImpl should be creatable", ex);
+            }
+            cf = ClassFilter.DEFAULT;
+        }
+
+        //TODO: It checks abstract classes, but not implementation
```

Comment:
The code below is specific  to Remoting serialization ONLY, because abstract classes are checked. It's not important for XStream. Ideally this code should be moved to JTH somehow

Reply:
-1